From bc26cebea4d82f7182b2c7a3a7c9318f4f0a9307 Mon Sep 17 00:00:00 2001
From: Tobias Schlemmer <keinstein@users.sourceforge.net>
Date: Wed, 24 Jan 2018 09:31:57 +0100
Subject: readline70-003
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Readline-Release: 7.0
Patch-ID: readline70-003

Bug-Reported-by:        Frédéric Brière <fbriere@fbriere.net>
Bug-Reference-ID:       <20170120180724.7ydq7fb2hsp366dj@fabul.fbriere.net>
Bug-Reference-URL:      http://lists.gnu.org/archive/html/bug-readline/2017-01/msg00002.html

Bug-Description:

Readline-7.0 uses pselect(2) to allow readline to handle signals that do not
interrupt read(2), such as SIGALRM, before reading another character.  The
signal mask used in the pselect call did not take into account signals the
calling application blocked before calling readline().
---
 input.c    | 1 +
 patchlevel | 2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/input.c b/input.c
index 286897d..24126ea 100644
--- a/input.c
+++ b/input.c
@@ -513,6 +513,7 @@ rl_getc (stream)
       result = 0;
 #if defined (HAVE_PSELECT)
       sigemptyset (&empty_set);
+      sigprocmask (SIG_BLOCK, (sigset_t *)NULL, &empty_set);
       FD_ZERO (&readfds);
       FD_SET (fileno (stream), &readfds);
       result = pselect (fileno (stream) + 1, &readfds, NULL, NULL, NULL, &empty_set);
diff --git a/patchlevel b/patchlevel
index 7cbda82..ce3e355 100644
--- a/patchlevel
+++ b/patchlevel
@@ -1,3 +1,3 @@
 # Do not edit -- exists only for use by patch
 
-2
+3
